src/components/organisms/EditArticleMetadata.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import FormLabel from "../molecules/FormLabel";
+import TextInput from "../atoms/TextInput";
+import Button from "../atoms/Button";
+import ArticleRevisionService from "../../services/ArticleRevisionService";
+
+const Inputs = styled("div")`
+	${({ $loading }) => $loading && "opacity: 50%;"}
+`;
+
+class EditArticleMetadata extends Component {
+	revision = new ArticleRevisionService();
+	state = {
+		loading: false,
+		error: false,
+		inputs: {
+			title: "",
+			description: "",
+			tags: ""
+		}
+	}
+
+	handleSubmit = async event => {
+		event.preventDefault();
+
+		this.setState({
+			loading: true
+		});
+
+		try {
+			const { title, description, tags } = this.state.inputs;
+			const articleData = this.props.article;
+
+			if ([title, description, tags].includes("")) {
```

Comment:
When the component first loads an article, the revision for it will be `null` - which makes sense. This means however, that when the state is set in `componentDidMount` description and tags will be `undefined`. This means the error handling here won't be active and an API request will be sent.

```
description: revision?.description,
tags: revision?.tags
```